### ISSUE TITLE
PIVX unstable when building from depends directory

### ISSUE THREAD

**Issue by ianw1974** [PIVX-Project/PIVX#2145]
### Describe the issue

When building PIVX 5.0 using depends directory, PIVX binaries are unstable.

First of which, starting a sync or doing a reindex, will cause the chain to stop at 267403 and then PIVX becomes unresponsive for CLI commands, so cannot pivx-cli stop, pivx-cli getinfo, etc, etc.  In a QT wallet, it causes DB corruption during the sync process, tried twice, once around 1.3M blocks, second at 1.5M blocks.  Chain stops syncing and segmentation faults if you attempt to re-run using the chain it attempted to sync.  With a successful chain put in .pivx directory and running same QT wallet, segmentation faults.

Second, if using a chain that is higher than 267403, the wallet starts fine but stops at "RPC in warm-up, verifying blocks".  This should normally only appear for a short time, but it was still at this point after 5 hours.  With QT wallet, segmentation faults.

Utilising the PIVX binaries on the release page didn't exhibit this behaviour, nor when we installed libstd-rust-dev and cargo packages and compiled without using the depends directory.

#### If so, please list the steps to reproduce below:
1. Clone PIVX repo
2. git checkout 5.0
3. Run commands as below to use depends directory:

```
./autogen.sh
./configure --disable-tests --disable-bench --prefix `pwd`/depends/x86_64-pc-linux-gnu
make
```

### Expected behavior
Binaries should compile and run without error

### Actual behavior
As above in "Describe the issue"

### What version of PIVX Core are you using?
Latest from branch 5.0.

### Machine specs:
- OS: Ubuntu 18.04
- CPU: Intel(R) Xeon(R) CPU E3-1270 v6 @ 3.80GHz
- RAM: 64GB
- Disk size: 2TB
- Disk Type (HD/SDD): HDD

**Comment by Fuzzbawls** [comment]
seems that you're conflating build/dependency issues with runtime issues. if your local chain is stuck at a particular block, the `debug.log` file will likely provide a reason as to why, which also likely won't have anything to do with build dependencies.

**Comment by ianw1974** [comment]
I don't think so, since when I build without using depends, everything works perfectly fine.  When I build using depends, that is when the chain doesn't work as expected.

For me it's not an issue, since I won't use the depends directory anymore, as it seems it doesn't work as expected for some reason, despite no compilation errors.  I have replicated this on two different systems when using the depends directory.

**Comment by ianw1974** [comment]
As extra info, the same chain I had on disk I used with the other binaries (compiled without depends), and it worked with the chain.  So it's not due to my chain locally being messed up, else both wouldn't sync.

**Comment by Fuzzbawls** [comment]
Please document, in verbatim, the steps you've taken to reproduce this then.

The official binaries are compiled using depends, and they work just fine. my own local depends-based builds also work without issue. Are you mixing depends and system dependencies? if so, that is always going to be cause for disaster and unknown behavior.

**Comment by ianw1974** [comment]
Sure no problem, please give me a short while to rule that out just in case.  I'll spin up a clean system to rule out any system dependencies like you mentioned and document the steps exactly.  And I'll document exactly the steps for both the compiles, as well as when I start using both types of compiled binaries.

**Comment by ianw1974** [comment]
After my tests, it is exactly how you wrote above - mixed system dependencies, so I will take that one on the chin, my fault for that one with my build environment.  Rather than run away and hide ;) I'll explain how we got to that situation.

So generally, I've been building without using the depends directory, until the new version just after the chain issue, an error appeared during compilation, so then built the other way on the same system.  I assumed since the build process finished that the binaries would be fine.  In our attempts to get fully synced, we had the symptoms as mentioned above - which as it seems instead of the binaries referring to the depends, they were utilising the system ones and thus library version mismatch (possibly) causing the instability.

As from my original post, we tried the release from pivx github, and it worked, and after further debugging last night, I reverted to the system way and realised I was missing libstd-rust-dev and cargo.  After installing these and compiling without depends it worked - even using the same chain snapshot that we had extracted earlier that constantly showed RPC warm-up in progress, verifying blocks.

Now, after building a completely new minimal install of Ubuntu 18.04, and just running ```apt install build-essential automake libtool pkg-config``` to get a minimal build environment, and building using depends directory:

```
CPUS=`cat /proc/cpuinfo | grep -ic proc`
cd depends
make -j$CPUS NO_QT=1
cd ..
./autogen.sh
./configure --without-gui --prefix `pwd`/depends/x86_64-pc-linux-gnu
make -j$CPUS
make install
cp depends/x86_64-pc-linux-gnu/bin/pivx* /usr/local/bin/
```

we were now able to run pivx-cli, pivxd without issue - this time the chain started successfully and downloaded blocks.  I used a chain snapshot from last night which was at blockheight 2684388.

So just posting the above for completeness, at least now it's clear to me what the issue was.  Thanks for the help and pointers in locating this.
